seismostats/analysis/b_significant.py

```diff
@@ -138,12 +141,13 @@ def transform_n(
     return b_transformed
 
 
-def bs_from_partitioning(
+def values_from_partitioning(
     list_magnitudes: list[np.ndarray],
     list_times: list[np.ndarray[np.datetime64]],
-    list_mc: float | np.ndarray,
+    list_mc: float | list[float],
```

Comment:
The "list of ndarrays" is fine-ish.

I would replace the np.datetime with normal datetime if it doesn't break anything - imo that is easier, and the user can still decide to try and pass np datetimes and it probably will work?`

what I don't like is that you call a variable "list_mc" and allow a float type. I see that it is more convenient to just be able to pass a single mc, but I think the API would be clearer if you just require all the lists to be of the same length, also easier to test.

Reply:
- Agree on the list[float]
- About the datetime: I think this is even not needed, because it also could be a list of floats, and it wouldn't be a problem. Going for list[np.ndarray]
- I think that the function becomes easier to use in some cases if a single mc is allowed.. also it is not really intended for the user, and if someone wants to use it, its actually well described in the docstrings. What do you think?

Reply:
we could have a long discussion on this point I think. In the end it is the difference between passing `np.full(len(list_magnitudes), mc)` to the function or having it in an if/else inside the function. whatever, as you want, still not happy about the name though ;P

nevertheless, to be consistent, you should check the sizeof list_mc if a list is passed...

Reply:
Ok, checked the size and even included a bonus of raining an IndexError :)